### ISSUE TITLE
mixing boolean and integers

### ISSUE THREAD

**Issue by noblige** [stefankoegl/python-json-patch#90]
It seems like `boolean` and `integer` types are not correctly compared, e.g.:
```
>>> print jsonpatch.make_patch({"field_1":True},{"field_2":1})
[{"path": "/field_2", "from": "/field_1", "op": "move"}]
```

```
>>> print jsonpatch.make_patch({"field":True},{"field":1})
[]
```

**Comment by noblige** [comment]
I feel this is incorrect https://github.com/stefankoegl/python-json-patch/blob/2990bb3d686c09a1550083336365e99247297711/jsonpatch.py#L820

```
        if src == dst:
            return
```
should this be something like:
```
        if src == dst and type(src) == type(dst):
            return
```

```
>>> a = 1
>>> b = True
>>> a == b
True
>>> a == b and type(a) == type(b)
False
>>> 
```

**Comment by stefankoegl** [comment]
This issue has been fixed in #106 which is included in version 1.25 that has just been released.
